spec/lib/whitehall_importer/migrate_assets_spec.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+RSpec.describe WhitehallImporter::MigrateAssets do
+  describe ".call" do
+    let(:asset_id) { "847150" }
+    let(:original_asset_url) { "https://asset-manager.gov.uk/blah/#{asset_id}/asset" }
+    let(:asset) { build(:whitehall_migration_asset_import) }
+
+    before :each do
+      stub_any_asset_manager_call
+    end
+
+    it "should take a WhitehallMigration::DocumentImport record as an argument" do
+      expect { described_class.call(build(:whitehall_migration_document_import)) }.not_to raise_error
+    end
+
+    it "should mark each asset as processing before its final state" do
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      expect(asset).to receive(:update!).with(state: "processing").once.ordered
+      expect(asset).to receive(:update!).with(hash_including(:state)).once.ordered
```

Comment:
I found this test a bit strange as we've got to get in pretty deep with the mocking and not test anything resultant. It made me wonder if we actually need this "processing" state - I'm not sure it really does anything for us?

Reply:
I would argue that it's worth setting the `processing` state so that, were we to attempt to run the migration task twice in quick succession, this wouldn't be part of the `migratable_assets` on the second run and we'd avoid attempting to migrate the same asset twice.

If it's worth setting the `processing` state, it's worth testing that it's set. Happy to remove though, as the thing it's protecting against is quite edge-casey.

Reply:
I don't think the consequences are too severe (are they actually negative? I guess it'd just do the same operation) were it to happen in quick succession so I think it's better to remove it and avoid needing quite deep tests for it.

Reply:
In that case do we want to remove the "processing" state altogether? It sounds like we have no real need for it - I wouldn't just want to remove the test but leave the application code in. Shall I go back through the commits and remove it from the enum, etc? (Happy to do so)

Reply:
That sounds a bit of a pain since there are 19 commits so I don't mind you adding a commit at the end which removes it - whatever you prefer.